spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/zuul/ZuulConfiguration.java

```diff
@@ -179,8 +186,38 @@ public ZuulRouteApplicationContextInitializer zuulRoutesApplicationContextInitia
 		private Map<String, ZuulFilter> filters;
 
 		@Bean
-		public ZuulFilterInitializer zuulFilterInitializer() {
-			return new ZuulFilterInitializer(this.filters);
+		public ZuulFilterInitializer zuulFilterInitializer(
+				CounterFactory counterFactory, TracerFactory tracerFactory) {
+			return new ZuulFilterInitializer(this.filters, counterFactory, tracerFactory);
+		}
+
+	}
+
+	@Configuration
+	protected static class ZuulEmptyMetricsConfiguration {
+
+		@ConditionalOnMissingBean(CounterFactory.class)
+		@Bean
+		public CounterFactory counterFactory() {
+			return new EmptyCounterFactory();
+		}
+
+		@ConditionalOnMissingBean(TracerFactory.class)
+		@Bean
+		public TracerFactory tracerFactory() {
+			return new EmptyTracerFactory();
+		}
+
+	}
+
+	@Configuration
+	protected static class ZuulMetricsConfiguration {
+
+		@ConditionalOnClass(CounterService.class)
```

Comment:
I'm not able to add `@ConditionalOnMissingClass("org.springframework.boot.actuate.metrics.CounterService")` to `ZuulEmptyMetricsConfiguration`, because it is present on classpath in tests. So none of 2 configurations get setup. To fix tests I would need to add `metrics-core` to classpath to enable `DropwizardAutoConfiguration`, so that `ZuulMetricsConfiguration` got setup.